cmd/proxy/actions/app.go

```diff
@@ -52,7 +52,6 @@ func App(logger *log.Logger, conf *config.Config) (http.Handler, error) {
 			SSLRedirect:     conf.ForceSSL,
 			SSLProxyHeaders: map[string]string{"X-Forwarded-Proto": "https"},
 		}).Handler,
-		mw.ContentType,
```

Comment:
Instead of removing this, perhaps adding a check, that is no `Content-Type` is set, then adding one. This would ensure there is always something present?

Reply:
I was going to do that, but I think all middle ware is executed before the route code.

To make matters more complicated, Set() doesn't seem to override the header, instead it's sent twice when set twice.

Reply:
Yeah, reasonable point. Then this middleware should be removed.

Reply:
I'm going to give this a second test run tonight, just to double check what I've told you here, but I'm 99% positive.

Reply:
I validated that what I stated is the case. The `Set()` method does not seem to override existing headers by the same name:

With `mw.ContentType`:
```
❯ curl -v http://localhost:3000/sumdb/sum.golang.org/lookup/github.com/matt0x6f/athens-test-repo@v1.0.0
*   Trying 127.0.0.1:3000...
* Connected to localhost (127.0.0.1) port 3000 (#0)
> GET /sumdb/sum.golang.org/lookup/github.com/matt0x6f/athens-test-repo@v1.0.0 HTTP/1.1
> Host: localhost:3000
> User-Agent: curl/7.81.0
> Accept: */*
> 
* Mark bundle as not supporting multiuse
< HTTP/1.1 200 OK
< Accept-Ranges: bytes
< Access-Control-Allow-Origin: *
< Alt-Svc: h3=":443"; ma=2592000,h3-29=":443"; ma=2592000
< Cache-Control: public, max-age=10800
< Content-Length: 389
< Content-Security-Policy-Report-Only: script-src 'none'; report-uri https://csp.withgoogle.com/csp/goa-fa2dfb7c
< Content-Type: application/json
< Content-Type: text/plain; charset=UTF-8
< Cross-Origin-Opener-Policy-Report-Only: same-origin; report-to="goa-fa2dfb7c"
< Date: Sun, 02 Jun 2024 19:15:50 GMT
< Expires: Sun, 02 Jun 2024 22:15:50 GMT
< Report-To: {"group":"goa-fa2dfb7c","max_age":2592000,"endpoints":[{"url":"https://csp.withgoogle.com/csp/report-to/goa-fa2dfb7c"}]}
< X-Content-Type-Options: nosniff
< X-Frame-Options: SAMEORIGIN
< X-Xss-Protection: 0
< 
25262111
github.com/matt0x6f/athens-test-repo v1.0.0 h1:QyR4h2t/LX1F5R9ipGD0Xy40Ti7wXWQtmZGb/ULfX4U=
github.com/matt0x6f/athens-test-repo v1.0.0/go.mod h1:AK0IEu9FDueopVcCOW9cuKpAZUymWO769GslenSoEzQ=

go.sum database tree
26548605
8GJ63plZadguUEd9PQ6EJwYIfgeb/nbtNfPhRA92Ux4=

— sum.golang.org Az3grsyr0VoqcgV66Tvivr6bW8C9xN6nIp6Rxx+jAoVEgtifXoJZElj+8OyZ7UUPMCTe0RKO0qSFK3Huc37fic2Rogk=
* Connection #0 to host localhost left intact
```

without:
```
❯ curl -v http://localhost:3000/sumdb/sum.golang.org/lookup/github.com/matt0x6f/athens-test-repo@v1.0.0
*   Trying 127.0.0.1:3000...
* Connected to localhost (127.0.0.1) port 3000 (#0)
> GET /sumdb/sum.golang.org/lookup/github.com/matt0x6f/athens-test-repo@v1.0.0 HTTP/1.1
> Host: localhost:3000
> User-Agent: curl/7.81.0
> Accept: */*
> 
* Mark bundle as not supporting multiuse
< HTTP/1.1 200 OK
< Accept-Ranges: bytes
< Access-Control-Allow-Origin: *
< Age: 194
< Alt-Svc: h3=":443"; ma=2592000,h3-29=":443"; ma=2592000
< Cache-Control: public, max-age=10800
< Content-Length: 389
< Content-Security-Policy-Report-Only: script-src 'none'; report-uri https://csp.withgoogle.com/csp/goa-fa2dfb7c
< Content-Type: text/plain; charset=UTF-8
< Cross-Origin-Opener-Policy-Report-Only: same-origin; report-to="goa-fa2dfb7c"
< Date: Sun, 02 Jun 2024 19:15:50 GMT
< Expires: Sun, 02 Jun 2024 22:15:50 GMT
< Report-To: {"group":"goa-fa2dfb7c","max_age":2592000,"endpoints":[{"url":"https://csp.withgoogle.com/csp/report-to/goa-fa2dfb7c"}]}
< X-Content-Type-Options: nosniff
< X-Frame-Options: SAMEORIGIN
< X-Xss-Protection: 0
< 
25262111
github.com/matt0x6f/athens-test-repo v1.0.0 h1:QyR4h2t/LX1F5R9ipGD0Xy40Ti7wXWQtmZGb/ULfX4U=
github.com/matt0x6f/athens-test-repo v1.0.0/go.mod h1:AK0IEu9FDueopVcCOW9cuKpAZUymWO769GslenSoEzQ=

go.sum database tree
26548605
8GJ63plZadguUEd9PQ6EJwYIfgeb/nbtNfPhRA92Ux4=

— sum.golang.org Az3grsyr0VoqcgV66Tvivr6bW8C9xN6nIp6Rxx+jAoVEgtifXoJZElj+8OyZ7UUPMCTe0RKO0qSFK3Huc37fic2Rogk=
* Connection #0 to host localhost left intact
```

We're looking for `text/plain` on this particular endpoint.

Reply:
https://github.com/golang/go/blob/master/src/net/url/url.go#L932

There is something else going on I think.

Reply:
Yeah, I agree, something is weird. When I click through what our `mw.ContentType` is referencing it uses the `response.Header()` method, which points to [this](https://github.com/golang/go/blob/master/src/net/http/header.go#L39). Same documentation as the URL header, but a different set of code.

I'm not really sure why it doesn't work as described. Both the middleware and the handler are working on the same API.